api/src/services/import-services/utils/row-state.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from 'zod';
+import { ApiGeneralError } from '../../../errors/api-error';
+import { CSVRow, CSVRowState } from '../../../utils/csv-utils/csv-config-validation.interface';
+
+/**
+ * Create a row state getter
+ *
+ * Note: This function allows both the row and the row state to be passed in.
+ *
+ * @example `
+ *  const getFromRowState = createRowStateGetter(z.object({ ... }));
+ *  getFromRowState(row)
+ *  getFromRowState(row[CSVRowState])
+ *  getFromRowState(row[CSVRowState].nestedState)
+ *  `
+ *
+ * @param {z.ZodSchema} schema - The Zod schema to validate the row state
+ * @returns {*} {function} - The row state getter
+ */
+export const createRowStateGetter = <SchemaType extends z.ZodSchema>(schema: SchemaType) => {
+  return (rowOrState: CSVRow | CSVRow[typeof CSVRowState]): z.infer<SchemaType> => {
+    // Check if the state or row is being passed
+    // Note: The row state is stored as a symbol on the row object
+    const isRow = Object.getOwnPropertySymbols(rowOrState).includes(CSVRowState);
+
+    // Get the state from the row
+    const state = isRow ? rowOrState?.[CSVRowState] : rowOrState;
+
```

Comment:
(optional) I think this type guard works (just to resolve the complaint about `rowOrState?.[CSVRowState]`).

```
const isRow = isCSVRow(rowOrState);
const state = isRow ? rowOrState[CSVRowState] : rowOrState;

export const isCSVRow = (rowOrState: CSVRow | CSVRow[typeof CSVRowState]): rowOrState is CSVRow => {
  return Object.getOwnPropertySymbols(rowOrState).includes(CSVRowState);
};
```

Reply:
This is another one of those cases where my editor isn't picking up on these warnings, so my apologies. (I still need to investigate why this is happening)

Fixed.

Reply:
Just let me know on your end if it is still present

Reply:
Seems happy on my end.